### PR TITLE
Put third parties in install dir and specify path to find them

### PR DESCRIPTION
**Pull request by julianoes**
Another find by @devbharat. Let me know if you agree @JonasVautherin.

### CODE REVIEW

File: src/MAVSDKConfig.cmake.in [hunk 1]
```diff
@@ -2,13 +2,13 @@ include(CMakeFindDependencyMacro)
 
 if(NOT @BUILD_SHARED_LIBS@)
     if(${CMAKE_VERSION} VERSION_LESS "3.9.0")
-        find_package(CURL REQUIRED CONFIG)
+        find_package(CURL REQUIRED CONFIG PATHS @DEPS_INSTALL_PATH@ NO_DEFAULT_PATH)
```
Comment (JonasVautherin) [inline src/MAVSDKConfig.cmake.in:5]: Doesn't `NO_DEFAULT_PATH` break everything that is not coming from our superbuild? What if I handled the dependencies manually and used `-DSUPERBUILD=OFF`? :thinking:

File: CMakeLists.txt [hunk 2]
```diff
@@ -49,7 +49,12 @@ if (BUILD_BACKEND)
 endif()
 
 set(DEPS_BUILD_PATH "${PROJECT_BINARY_DIR}/third_party" CACHE PATH "Install path for the dependencies. Ignored if SUPERBUILD=OFF.")
-set(DEPS_INSTALL_PATH "${DEPS_BUILD_PATH}/install" CACHE PATH "Install path for the dependencies. Ignored if SUPERBUILD=OFF.")
+
+if(NOT DEFINED CMAKE_INSTALL_PREFIX)
+  set(DEPS_INSTALL_PATH "${DEPS_BUILD_PATH}/install" CACHE PATH "Install path for the dependencies. Ignored if SUPERBUILD=OFF.")
+else()
+  set(DEPS_INSTALL_PATH "${CMAKE_INSTALL_PREFIX}/third_party/install" CACHE PATH "Install path for the dependencies. Ignored if SUPERBUILD=OFF.")
+endif()
```
Comment (JonasVautherin) [inline CMakeLists.txt:57]: Not sure if I am comfortable with this :sweat_smile:. This essentially changes the meaning of `CMAKE_INSTALL_PREFIX`, which is supposed to install MAVSDK (not MAVSDK + a bunch of libraries).

In fact, I would almost be in favor of removing the whole "automated superbuild" logic (i.e. keeping `third_party/`, but not running it at configure time) and documenting how to use it (as a "helper subproject" rather than a "superbuild", if I may say).

So building "with the superbuild" would be something like this:

```cmake
# Build the dependencies and install them locally in third_party/build
cmake -DCMAKE_INSTALL_PREFIX=/path/to/install/dependencies -Bthird_party/build -Sthird_party
cmake --build third_party/build

# Build MAVSDK, using the dependencies above
cmake -DCMAKE_PREFIX_PATH=/path/to/install/dependencies -Bbuild/default -S.
cmake --build build/default
```

In other words, that would make it less automated (and force the user to understand that there is a step where they need to think about the dependencies), and not more automated.

My usual fear with more automation is to slowly but safely move towards the PX4 build system, where every step was surely well-meant, but contributed making it convoluted and very custom.

What do you think?
